src/main/scala/SqliteDb.scala

```diff
@@ -77,15 +77,15 @@ class SqliteResultIterator(db: SqliteDb, private val stmt: Long)
                         case Sqlite3C.TEXT => SqlText(new String(Sqlite3C.column_blob(stmt, i)))
                         case Sqlite3C.BLOB => SqlBlob(Sqlite3C.column_blob(stmt, i))
                         case Sqlite3C.NULL => SqlNull
-                        case _ => sys.error("unsupported type")
+                        case _ => throw SqlException("unsupported type")
                     }
                 }
             case Sqlite3C.DONE =>
                 null
             case Sqlite3C.ERROR =>
-                sys.error("sqlite error: " + db.errmsg)
+                throw SqlException(s"sqlite error: ${db.errmsg}")
             case other =>
-                sys.error("unexpected result: " + other)
+                throw SqlException(s"${db.errmsg}\nresult code: $other (${SqliteDb.resultToMessage(other)})")
```

Comment:
According to my version of this commit, I seemed to believe that `db.errmsg` could be empty. Not sure if that matters, but it would make for some funny formatting...